### ISSUE TITLE
Would be nice for the Result type to have a `unwrap` method.

### ISSUE THREAD

**Issue by gluax** [true-myth/true-myth#817]
It would be excellent for the `Result` type to have the `unwrap` method matching `rust`.

I like using it in tests, where I know the result should be ok.

```ts
const res = some_function_that_returns_a_result();

// currently I do 
expect(res.isOk).toBe(true);
//Then I have to awkwardly extract it. I already know it's `OK`.

//whereas in Rust, I could do:
const res = some_function_that_returns_a_result().unwrap();
// Would throw the error if it's not an error. But that's ok cause this is a test situation where I expect this to succeed.
```

It's just a nice QOL use case, especially in testing. On the same vein supporting `unwrap_err` would also be nice :)

**Comment by chriskrycho** [comment]
I think I have thought about adding this back in about seven times. 😂 It makes a lot of sense—so much so that we actually *used* to have it, under the names `unsafelyUnwrap` and `unsafelyUnwrapErr`. We removed it back in 2022 when we made it possible to just use those value accessors, but the test scenario is definitely the place that is *least* convenient. Let us mull on this a bit—it’s not super hard to add back in, of course, just a question whether we want to do that or maybe provide helpers in a different way oriented specifically at tests or something.

My reasoning for considering the latter rather than just adding them directly is twofold:

1. Tree-shaking is good, but it is not perfect, especially when dealing with object types like the ones we use here. Given how bundling works, a method is much less likely to get stripped than a standalone function is, and in some cases *cannot* be. I am sensitive to the fact that this is a *small* but not a *tiny* library, and in many cases every byte counts! So adding back in the methods is small but I do actually want to think about it.
2. I think at a philosophical level, in practice I *want* folks to actually do the runtime check—particularly since it is not meaningfully more expensive than doing `unwrap` and in fact is only *safe* to do in in the case that you *have* done the check. Against that, though, I balance the desire to have test code look “the same” as runtime code as much possible.

Net, let us think on it and chat a bit and we’ll make a call!

> [!NOTE]
> If we added it back to `Result`, we would want to add it back to `Maybe` as well!

**Comment by gluax** [comment]
The more I have to work with true-myth in a test environment, the lack of a way to just unwrap something after I assert `isJust`/I know from a test perspective should never be `Nothing`, is starting to drive me crazy 😅.

**Comment by chriskrycho** [comment]
@gluax I hear you! How does this feel?

```ts
import Maybe from 'true-myth/maybe';
import { unwrap, unwrapErr } from 'true-myth/toolbelt';
import { expect, test } from 'vitest'; // etc.

import { producesMaybe, producesResult } from 'my-library';

test('using this new helper', () => {
  expect(unwrap(producesMaybe())).toBe(true);
  expect(unwrap(producesResult('valid'))).toBe('cool');
  expect(unwrapErr(producesResult('invalid')).toBe('oh teh noes');
});
```

I *believe* that should be straightforward to implement, and keeps things nicely separate in terms of bundle implications. It’s very mildly less nice than doing `producesMaybe().unwrap()`, but in my judgment not massively so, and not more so than a lot of other ways we approach test-specific code. What do you think? (cc. @bmakuh as well!)

**Comment by chriskrycho** [comment]
It could also work to add them to a `true-myth/test-support` module, which would even *more* loudly signal that these are really only intended for test code—with nothing *stopping* you from adding them to non-test code, but with a very loud signal in the name (and therefore an easy lint rule etc.) that they are really only for testing!

**Comment by chriskrycho** [comment]
That’s actually exactly what I ended up doing; see #870 – I’ll go ahead and merge and release it once there’s been time for a bit of feedback!

**Comment by chriskrycho** [comment]
Released as part of [v8.1.0](https://github.com/true-myth/true-myth/releases/tag/v8.1.0)!
